packages/react-reconciler/src/ReactFiberCompleteWork.js

```diff
@@ -588,15 +588,18 @@ function preloadResourceAndSuspendIfNeeded(
 
   workInProgress.flags |= MaySuspendCommit;
 
-  const rootRenderLanes = getWorkInProgressRootRenderLanes();
-  if (!includesOnlyNonUrgentLanes(rootRenderLanes)) {
-    // This is an urgent render. Don't suspend or show a fallback.
-  } else {
-    const isReady = preloadResource(resource);
-    if (!isReady) {
-      if (shouldRemainOnPreviousScreen()) {
-        workInProgress.flags |= ShouldSuspendCommit;
-      } else {
+  const {ready, required} = preloadResource(resource);
+  if (!ready) {
+    if (shouldRemainOnPreviousScreen()) {
+      // We can stay on the current screen until the resource is ready
+      // so we mark this fiber as suspending the commit
+      workInProgress.flags |= ShouldSuspendCommit;
+    } else {
+      const rootRenderLanes = getWorkInProgressRootRenderLanes();
+      if (!includesOnlyNonUrgentLanes(rootRenderLanes) && required) {
```

Comment:
I don't really like introducing this concept of required and not-required resources. It also only gets read when you're rendering a new tree, not during an update; when you rendering a new tree, it's pretty much always fine to show a fallback, even if the render is synchronous.

The case where you might want to distinguish between "required" and "not required" resources is during a sync update, when the choice is between a stale resource or replacing the existing UI with a fallback (bad).

But that can be modeled during the commit phase with a timeout of effectively 0. I think that's the better strategy.